pkg/network/protocols/http/shared_libraries_test.go

```diff
@@ -64,11 +71,66 @@ func TestSharedLibraryDetection(t *testing.T) {
 	assert.Equal(t, fpath, pathDetected)
 }
 
+func TestSharedLibraryDetectionWithRoot(t *testing.T) {
+	t.Skip("skip for the moment as some distro are not friendly with busybox package")
+
+	tdir := t.TempDir()
+	root := filepath.Join(tdir, "root")
+	err := os.MkdirAll(root, 0755)
+	require.NoError(t, err)
+
+	libpath := "/fooroot.so"
+
+	simulateOpenAt(root + libpath)
+	err = exec.Command("cp", "/usr/bin/busybox", root+"/ash").Run()
+	require.NoError(t, err)
+	err = exec.Command("cp", "/usr/bin/busybox", root+"/sleep").Run()
+	require.NoError(t, err)
+
+	perfHandler, doneFn := initEBPFProgram(t)
+	t.Cleanup(doneFn)
+
+	var (
+		mux          sync.Mutex
+		pathDetected string
+	)
+
+	callback := func(id pathIdentifier, root string, path string) error {
+		mux.Lock()
+		defer mux.Unlock()
+		pathDetected = path
+		return nil
+	}
+
+	watcher := newSOWatcher(perfHandler,
+		soRule{
+			re:         regexp.MustCompile(`fooroot.so`),
+			registerCB: callback,
+		},
+	)
+	watcher.Start()
+
+	time.Sleep(10 * time.Millisecond)
+	//	simulateOpenAt(fpath)
+	o, err := exec.Command("unshare", "--fork", "--pid", "-R", root, "/ash", "-c", fmt.Sprintf("sleep 1 > %s", libpath)).CombinedOutput()
+	t.Log(string(o))
+	assert.NoError(t, err)
```

Comment:
you are mixing between `require.NoError` and `assert.NoError`
please stick with one of them